src/parser/shared/modules/Enemies.ts

```diff
@@ -72,6 +73,42 @@ class Enemies extends Entities<Enemy> {
     return enemy;
   }
 
+  /**
+   * Retrieves the enemy that was the source of given event.
+   * @param {object} event
+   * @returns {Enemy|null}
+   */
+  getSourceEntity(event: AnyEvent): Enemy | null {
+    if (!HasSource(event)) {
+      return null;
+    }
+    if (event.sourceIsFriendly) {
+      return null;
+    }
+    if (!event.sourceID) {
+      return null;
+    }
```

Comment:
`!HasSource(event)` and `!event.sourceID` are redundant

Reply:
Also makes the `"sourceInstance" in event` check unnecessary